fixups/RegLegacyFixups/InitializeFixup.cpp

```diff
@@ -117,12 +121,102 @@ void LogString(const wchar_t* name, const wchar_t* value)
     Log(L"%ls=%ls\n", name, value);
 }
 
+struct EntryToCreate 
+{
+    std::wstring path;
+    std::unordered_map<std::wstring, std::wstring> values;
+};
+
 
-void InitializeFixups()
+void CreateRegistryRedirectEntries(std::wstring_view dependency_name, std::vector<EntryToCreate> entries)
 {
+    Log("RegLegacyFixups Create redirected registry values\n");
+
+    Package pkg = Package::Current();
+    IVectorView<Package> dependencies = pkg.Dependencies();
+
+    std::wregex dependency_version_regex = std::wregex(L"%dependency_version%");
+    std::wregex dependency_path_regex = std::wregex(L"%dependency_root_path%");
+
+    Package dependency = nullptr;
+
+#ifdef _DEBUG
+    Log("Filtering required dependency\n");
+#endif
+    for (auto&& dep : dependencies) {
+        std::wstring_view name = dep.Id().Name();
+#ifdef _DEBUG
+        Log("Checking package dependency %.*LS\n", name.length(), name.data());
+#endif
+
+        if (name.find(dependency_name) != std::wstring::npos) {
+#ifdef _DEBUG
+            Log("Found required dependency\n");
+#endif
+            dependency = dep;
+            break;
+        }
+    }
+    if (dependency == nullptr) 
```

Comment:
If `dependency` is a pointer please remove `== nullptr`

Reply:
Updated